compiler/rustc_middle/src/ty/adt.rs

```diff
@@ -90,11 +90,11 @@ pub struct AdtDefData {
     /// The `DefId` of the struct, enum or union item.
     pub did: DefId,
     /// Variants of the ADT. If this is a struct or union, then there will be a single variant.
-    variants: IndexVec<VariantIdx, VariantDef>,
+    pub(crate) variants: IndexVec<VariantIdx, VariantDef>,
     /// Flags of the ADT (e.g., is this a struct? is this non-exhaustive?).
-    flags: AdtFlags,
+    pub(crate) flags: AdtFlags,
     /// Repr options provided by the user.
-    repr: ReprOptions,
+    pub(crate) repr: ReprOptions,
```

Comment:
These should probably be accessed through [`AdtDef::flags`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.AdtDef.html#method.flags), etc.